src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,57 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DATA_CORRECT_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
+    private static final int MAXIMUM_AGE = 100;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        throwNullException(user, "User");
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        checkAge(user.getAge());
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkAge(int age) {
+        if (age < MINIMUM_AGE) {
+            throw new FailedRegistrationException("Dear user, you're too young. "
+                    + "Please, come back in " + (MINIMUM_AGE - age));
+        }
+        if (age > MAXIMUM_AGE) {
+            throw new FailedRegistrationException("Dear user, we can't believe,"
+                    + " that you are so old.");
+        }
+    }
+
+    private void checkPassword(String password) {
+        throwNullException(password, "Password");
+        if (password.length() < DATA_CORRECT_LENGTH) {
+            throw new FailedRegistrationException(getExceptionMessage(password, "Password"));
+        }
+    }
+
+    private void checkLogin(String login) {
+        throwNullException(login, "Login");
+        if (login.length() < DATA_CORRECT_LENGTH) {
```

Comment:
what will happen if login or password will be blank like `      `

you should cover that as well